internal/convert/yaml.go

```diff
@@ -17,11 +19,17 @@ func YAMLToMapOfInterfaces(input string) (map[interface{}]interface{}, error) {
 func YAMLSliceOfInterfaceToSliceOfMaps(input []interface{}) ([]map[interface{}]interface{}, error) {
 	output := make([]map[interface{}]interface{}, 0)
 	for _, current := range input {
-		data, err := YAMLToMapOfInterfaces(current.(string))
-		if err != nil {
-			return nil, err
+		// Apply YAMLToMap only if string is passed
+		switch c := current.(type) {
+			case string:
+				data, err := YAMLToMapOfInterfaces(c)
```

Comment:
```suggestion
				data, err := YAMLToMapOfInterfaces(current.(string))
```

@martinhaus why sending the type to the function instead of the value?

also, the check could be done easier:

```
if currentYaml, ok := current.(string); ok {
       data, err := YAMLToMapOfInterfaces(currentYaml)
    }
```

Thank you very much for the contribution.

Reply:
`c` should actually be the same as `current.(string)` in this case. Explained [here](https://stackoverflow.com/a/6373523/6456319).

For the second part - yes, your way looks clearer. I can change it if you prefer it that way.